### ISSUE TITLE
Does not compile without `serde` feature

### ISSUE THREAD

**Issue by DragonDev1906** [ralexstokes/ethereum-consensus#214]
There is a PR that tried to fix this, but it was not merged yet and I don't know if it is still current: #191

On the current main branch `ethereum-consensus` does not compile without the `serde` feature due to the reason stated in #191:
```bash
cargo check --no-default-features
```

**Comment by ralexstokes** [comment]
thanks for raising this, I think the changes in #191 make sense but also make the code more dense than it already is, so if anything I'd possibly even just support removing the `serde` feature and having the library support it by default

are you doing something where this is blocking you?

**Comment by DragonDev1906** [comment]
> are you doing something where this is blocking you?

No, I'm likely going to need serde anyways. Even if I don't the main downside is longer compile times.
For me `--no-default-features -F serde` is sufficient.

I just wanted to mention that the `serde` feature doesn't work as expected.

**Comment by ralexstokes** [comment]
great, thanks for the message :)

I'll leave this open, and (1) if anyone runs into this as blocker please post here and (2) ill keep this in my queue to address sooner or later
